CMake/ClangCompilers.cmake

```diff
@@ -19,6 +19,10 @@ if(QMC_OMP)
         CACHE STRING "Offload target architecture")
     set(OPENMP_OFFLOAD_COMPILE_OPTIONS "-fopenmp-targets=${OFFLOAD_TARGET}")
 
+    if(NOT DEFINED OFFLOAD_ARCH AND OFFLOAD_TARGET MATCHES "nvptx64" AND DEFINED CMAKE_CUDA_ARCHITECTURES)
+      set(OFFLOAD_ARCH sm_${CMAKE_CUDA_ARCHITECTURES})
```

Comment:
@ye-luo Do we allow `CMAKE_CUDA_ARCHITECTURES` be a list of integers?

Reply:
CMAKE_CUDA_ARCHITECTURES should be allowed as the name indicates.
LLVM doesn't support offload to multiple architectures but one.
We might need to skip this fallback if CMAKE_CUDA_ARCHITECTURES is a list of integers. Is this the check you would like to add?

Reply:
I added a check that the list length is 1

Reply:
I also made NVHPC working with CMAKE_CUDA_ARCHITECTURES  as a list of integers.